File: solutions/csharp/CoffeeVendingMachine/Coffee.cs
namespace CoffeeVendingMachine;

public class Coffee
{
    private readonly string _name;
    private readonly double _price;
    private readonly Dictionary<Ingredient, int> _recipe;

    public Coffee(string name, double price, Dictionary<Ingredient, int> recipe)
    {
        _name = name;
        _price = price;
        _recipe = recipe;
    }
    
    public string GetName() => _name;
    
    public double GetPrice() => _price;
    
    public Dictionary<Ingredient, int> GetRecipe() => _recipe;
}

File: solutions/golang/coffeevendingmachine/coffee.go
package coffeevendingmachine

type Coffee struct {
	name   string
	price  float64
	recipe map[*Ingredient]int
}

func NewCoffee(name string, price float64, recipe map[*Ingredient]int) *Coffee {
	return &Coffee{
		name:   name,
		price:  price,
		recipe: recipe,
	}
}

func (c *Coffee) GetName() string {
	return c.name
}

func (c *Coffee) GetPrice() float64 {
	return c.price
}

func (c *Coffee) GetRecipe() map[*Ingredient]int {
	return c.recipe
}
